### ISSUE TITLE
train loss

### ISSUE THREAD

**Issue by TonyUSTC** [serdarozsoy/corinfomax-ssl#1]
thanks for your work。
I try use corinfomax loss for SSL speaker  recognition task, but my train loss will suddenly rise at end of each epoch。
can anyone tell me why ?
<img width="1480" alt="image" src="https://user-images.githubusercontent.com/26316926/192714501-cd095c6b-00f8-4711-a0ca-9a7aee0e1ac6.png">

**Comment by serdarozsoy** [comment]
Thank you for your interest.

As you are adopting our code for a different data set (speech), it is hard to predict the actual reason for such a behavior with these details. However, the following might be useful tips for your analysis:

1 - I guess the plot you shared has global steps in its x-axis, not epoch averaged as in the original code. Do these "bumps" appear at the last step of each epoch or in the different steps of each epoch? As I see, bumps do not seem periodic and smoothed loss decreases, so these bumps may depend on sample distributions of batches.

2 - In addition, these bumps may be dependent on the hyperparameter selections, including augmentations, for your dataset. Initially, you can try lowering the learning rate. 

3 - If the bumps are happening at the last step of each epoch, you can check "drop_last" in [dataloader](https://github.com/serdarozsoy/corinfomax-ssl/blob/1bea4f972f4e1d9a340880b08e68531c613671dd/cifars-tiny/main_traintest.py#L30). It should be "True" for pretrain dataset. 

4 - Similarly, you can check "shuffle" in [dataloader](https://github.com/serdarozsoy/corinfomax-ssl/blob/1bea4f972f4e1d9a340880b08e68531c613671dd/cifars-tiny/main_traintest.py#L30). It should be "True" for pretrain dataset.

5 - Changing code from epochs to global steps may cause a reporting inconsistency issue for losses in original code, especially if it is a multigpu implementation.

**Comment by serdarozsoy** [comment]
This issue was closed as it has not been active for 2 months since the last reply.
